scripts/local_k8s/unittests/test_cli.py

```
@@ -0,0 +1,157 @@
+        encoding="utf8",
+    )
+    fake_tool.chmod(0o755)
+    for tool_name in ("docker", "helm", "k3d", "kubectl"):
+        (fake_bin / tool_name).symlink_to(fake_tool)
+
+
+def _run_make_targets(env: dict[str, str], targets: tuple[str, ...]) -> None:
+    """Run preview Makefile targets through the real CLI boundary."""
+    for target in targets:
+        completed = subprocess.run(  # noqa: S603 - argv is fixed by the test.
+            ["make", "--no-print-directory", target],
```

Ruff (0.15.20)

[error] Starting a process with a partial executable path

(S607)